### PR TITLE
Small test e2e updates / boilerplate notes

### PR DESCRIPTION
**Pull request by ValarDragon**
## What is the purpose of the change

Some small things I noticed on reading the code, in addition to issues I'm making on the side!

## Brief Changelog

* Being a human linter
* Add some TODO's
* introduce a `NewGenesisEpochInfo` function, and use to eliminate overhead.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: tests/e2e/e2e_util_test.go [hunk 1]
```diff
@@ -61,10 +61,6 @@ func (s *IntegrationTestSuite) ExecTx(chainId string, validatorIndex int, comman
 				return false
 			}
 
-			if err != nil {
-				return false
-			}
```
Comment (ValarDragon) [inline tests/e2e/e2e_util_test.go:66]: Duplicate of lines 60-62

**Review by czarcas7ic — APPROVED** [review]
Thanks so much for cleaning up some of the mess I made! All looks good to me, one thing though; can you further explain what you are expecting in regards to this message https://github.com/osmosis-labs/osmosis/issues/1623#issuecomment-1141538814, I wasn't quite following.

**Comment by ValarDragon** [comment]
You did a great job on these! Not really a mess here at all. Don't expect to build things perfectly the first time lol, getting something that works, and then iterating to improve it is the way. A lot of this highlights general improvements we need to make to the code overall!

For that particular comment, we see in `s.ExecTx(config.meta.Id, 0, cmd, "code: 0")` , "code: 0" as the sentinel for correctness in ~every call. Can just replace that with a bool for do we expect the tx to succeed vs fail. (Still need to provide our own sentinel for "ExecCLI")

**Comment by czarcas7ic** [comment]
If thats the case though, we cannot use this function for hermes as well, which looks for a "Success" string or something along those lines instead of a "code: 0". So I think that is why I opted for the string so I could use it across the two. If this is not good practice though, would be happy to implement your suggestion!

**Comment by czarcas7ic** [comment]
Here https://github.com/osmosis-labs/osmosis/blob/main/tests/e2e/e2e_util_test.go#L116
and here https://github.com/osmosis-labs/osmosis/blob/main/tests/e2e/e2e_util_test.go#L122

**Comment by ValarDragon** [comment]
Those Hermes calls would be calls to `ExecCLI` not `ExecTx` !

**Comment by czarcas7ic** [comment]
Gotcha, makes sense now, thanks!

File: tests/e2e/chain/config.go [hunk 2]
```diff
@@ -86,6 +87,7 @@ func addAccount(path, moniker, amountStr string, accAddr sdk.AccAddress) error {
 	balances := banktypes.Balance{Address: accAddr.String(), Coins: coins.Sort()}
 	genAccount := authtypes.NewBaseAccount(accAddr, nil, 0, 0)
 
+	// TODO: Make the SDK make it far cleaner to add an account to GenesisState
```
Comment (p0mvn) [inline tests/e2e/chain/config.go:90]: Going to create an issue instead
```suggestion
```

**Review by p0mvn — APPROVED** [review]

File: tests/e2e/chain/config.go (hunk 2 above)
Comment (p0mvn) [reply tests/e2e/chain/config.go:90]: https://github.com/osmosis-labs/osmosis/issues/1628

Comment (ValarDragon) [reply tests/e2e/chain/config.go:90]: I made one on the sdk earlier

Comment (ValarDragon) [reply tests/e2e/chain/config.go:90]: Also tracked here https://github.com/cosmos/cosmos-sdk/issues/12091

File: tests/e2e/chain/validator.go [hunk 3]
```diff
@@ -270,8 +272,9 @@ func (v *internalValidator) signMsg(msgs ...sdk.Msg) (*sdktx.Tx, error) {
 
 	txBuilder.SetMemo(fmt.Sprintf("%s@%s:26656", v.nodeKey.ID(), v.instanceName()))
 	txBuilder.SetFeeAmount(sdk.NewCoins())
-	txBuilder.SetGasLimit(200000)
+	txBuilder.SetGasLimit(uint64(200000 * len(msgs)))
```
[the remark this answers is not in the document]
Comment (ValarDragon) [reply tests/e2e/chain/validator.go:275]: If we do multi-messages gas consumption will increase notably and the limit won't suffice.

**Pull request merged**